src/main/java/org/opensearch/neuralsearch/search/query/HybridCollectorManager.java

```diff
@@ -60,27 +69,44 @@ public static CollectorManager createHybridCollectorManager(final SearchContext
         int numDocs = Math.min(searchContext.from() + searchContext.size(), totalNumDocs);
         int trackTotalHitsUpTo = searchContext.trackTotalHitsUpTo();
 
+        Weight filteringWeight = null;
+        // Check for post filter to create weight for filter query and later use that weight in the search workflow
+        if (Objects.nonNull(searchContext.parsedPostFilter()) && Objects.nonNull(searchContext.parsedPostFilter().query())) {
+            Query filterQuery = searchContext.parsedPostFilter().query();
+            ContextIndexSearcher searcher = searchContext.searcher();
+            // ScoreMode COMPLETE_NO_SCORES will be passed as post_filter does not contribute in scoring. COMPLETE_NO_SCORES means it is not
+            // a scoring clause
+            // Boost factor 1f is taken because if boost is multiplicative of 1 then it means "no boost"
+            // Previously this code in OpenSearch looked like
+            // https://github.com/opensearch-project/OpenSearch/commit/36a5cf8f35e5cbaa1ff857b5a5db8c02edc1a187
+            filteringWeight = searcher.createWeight(searcher.rewrite(filterQuery), ScoreMode.COMPLETE_NO_SCORES, boost_factor);
+        }
+
         return searchContext.shouldUseConcurrentSearch()
             ? new HybridCollectorConcurrentSearchManager(
                 numDocs,
                 new HitsThresholdChecker(Math.max(numDocs, searchContext.trackTotalHitsUpTo())),
                 isSingleShard,
                 trackTotalHitsUpTo,
-                searchContext.sort()
+                searchContext.sort(),
+                filteringWeight
             )
             : new HybridCollectorNonConcurrentManager(
                 numDocs,
                 new HitsThresholdChecker(Math.max(numDocs, searchContext.trackTotalHitsUpTo())),
                 isSingleShard,
                 trackTotalHitsUpTo,
-                searchContext.sort()
+                searchContext.sort(),
+                filteringWeight
             );
     }
 
     @Override
     public Collector newCollector() {
         Collector hybridcollector = new HybridTopScoreDocCollector(numHits, hitsThresholdChecker);
-        return hybridcollector;
+        // Check if filterWeight is present. If it is present then return wrap Hybrid collector object underneath the FilteredCollector
+        // object and return it.
+        return filterWeight == null ? hybridcollector : new FilteredCollector(hybridcollector, filterWeight);
```

Comment:
for same checks we used Objects.isNull is other places in the code

Reply:
As discussed with @navneet1v, this static call is call made to Objects class for just a simple check. Therefore, we decided to go with simple null check. Initially my plan was to add Objects.nonNull check but @navneet1v insisted to use simple check to avoid extra call.

Reply:
@vibrantvarun lets make things consistent in this PR for now.